woocommerce/woocommerce
# Implement ProductRenderer class for rendering and interactivity

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR refactors the class Product Collection Controller (ex Product Collection) in the following way:
- Some methods were moved from `ProductCollectionController` to `ProductRenderer `.

This change is part of a bigger change described in [this epic](https://github.com/woocommerce/woocommerce/issues/52418).

Closes #52416.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch 
2. Ensure `ProductCollectionController` (ex `ProductCollection`) utilizes `ProductRenderer ` for all rendering operations.
3. Verify that all existing functionalities continue to work as expected after the refactor.
4. Test all the Product Collection types.
6. Verify that all the tests pass correctly.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
